group_by.py

```diff
@@ -0,0 +1,12 @@
+def group_by(func, iterable):
```

Comment:
Use type hints and return types -
https://www.geeksforgeeks.org/duck-typing-in-python/

Reply:
Done